### ISSUE TITLE
Choose a license

### ISSUE THREAD

**Issue by smichel17** [LibreShift/TransitionScheduler#1]
I'm generally a fan of strong copyleft licenses (eg, GPL/AGPL), but for a library like this LGPL or MIT could make sense, too. Any opinions?

**Comment by smichel17** [comment]
@corphish @fython @joonatoona

**Comment by joonatoona** [comment]
As far as I understand it, the LGPL is identical, except that dynamically linking doesn't constitute a derivative work. I honestly have no clue how that applies to Java, which is probably why people choose the GPL with an exception.

**Comment by smichel17** [comment]
I wasn't really sure, so I looked it up. The most understandable explanation I found was question 6 [here](https://resources.whitesourcesoftware.com/blog-whitesource/top-9-gpl-with-the-classpath-exception-questions-answered), which says the LGPL has some additional protections for end users (allows them to reverse engineer the derivative software, modify the library portion, and then run the derivative using their modified version of the library). Someone on [stackexchange](https://softwareengineering.stackexchange.com/questions/119436/what-does-gpl-with-classpath-exception-mean-in-practice/326325) thinks the LGPL is unclear when applied to java; the fsf [disagrees](https://www.gnu.org/licenses/lgpl-java.html).

In form, the LGPL used to be a separate license, but as of v3, the [license text](https://www.gnu.org/licenses/lgpl-3.0.en.html) is just an addition to the GPLv3, the same way the classpath exception is set up (obviously with different terms).

---

In the end, I'm inclined to just go with GPL-3.0+, put a note in the README asking people to open an issue if they want to use the lib in a proprietary app, and figure out what to do if anyone actually opens an issue (I suspect mostly it will only be the 3-4 of our apps that use the library, anyway).
